internal/storage/image.go

```diff
@@ -398,6 +408,145 @@ func (svc *imageService) ImageStatus(systemContext *types.SystemContext, nameOrI
 	return &result, nil
 }
 
+// nolint:staticcheck
+func (svc *imageService) IsRunningImageAllowed(ctx context.Context, systemContext *types.SystemContext, userSpecifiedImage string) error {
+	userSpecifiedImageParsed, err := reference.ParseNormalizedNamed(userSpecifiedImage)
+	if err != nil {
+		return fmt.Errorf("parsing user specified image: %w", err)
+	}
+	userSpecifiedImageParsed = reference.TagNameOnly(userSpecifiedImageParsed)
+
+	policy, err := signature.DefaultPolicy(systemContext)
+	if err != nil {
+		return fmt.Errorf("get default policy: %w", err)
+	}
+
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return fmt.Errorf("create policy context: %w", err)
+	}
+
+	localManifestDigest, err := svc.pullLocalManifestDigest(ctx, systemContext, policyContext, userSpecifiedImageParsed)
+	if err != nil {
+		return fmt.Errorf("pull local manifest digest: %w", err)
+	}
+	if localManifestDigest == "" {
+		return nil
+	}
+	log.Debugf(ctx, "Local manifest digest is: %s", localManifestDigest)
+
+	if err := svc.checkSignature(ctx, systemContext, policyContext, userSpecifiedImageParsed, localManifestDigest); err != nil {
+		return fmt.Errorf("checking signature of %q: %w", userSpecifiedImageParsed.String(), err)
+	}
+
+	return nil
+}
+
+func (svc *imageService) pullLocalManifestDigest(ctx context.Context, sys *types.SystemContext, policyContext *signature.PolicyContext, userSpecifiedImage reference.Named) (digest.Digest, error) {
+	srcRef, err := docker.NewReference(userSpecifiedImage)
+	if err != nil {
+		return "", fmt.Errorf("creating docker:// reference for %q: %w", userSpecifiedImage.String(), err)
+	}
+
+	destRef, err := istorage.Transport.NewStoreReference(svc.store, userSpecifiedImage, "")
+	if err != nil {
+		return "", fmt.Errorf("creating containers-storage: reference for %q: %w", userSpecifiedImage.String(), err)
+	}
+
+	manifestBytes, err := copy.Image(ctx, policyContext, destRef, srcRef, &copy.Options{
```

Comment:
Without credentials this doesn’t work for authenticated registries at all.

The prototype’s `manifestBytes` was intended to be collected in `PullImage` and to be used to form the correct `RepoDigests` value.